### DISCUSSION TITLE
A manual 'update' method for signals.

### DISCUSSION THREAD

**Discussion by flmng0** [preactjs/signals#376] [Ideas]
Hi everyone.

I know this is an issue with Proxies and not signals specifically, but having to do:
```js
todos.value.splice(index, 1);
todos.value = [...todos.value]; // <--- this
```

is not favourable.

I understand the why, however, I don't see why something like `Signal.update()` couldn't be implemented.

If this is silly, I apologise, I haven't looked at the source for `signals-core` just yet, but I would assume this is pretty simple to do.

**Comment by rschristian** [comment]
Looks to be similar to #353.

While it's likely simple to implement, it grows the API surface area and creates multiple ways of setting a new signal value, which isn't really ideal.

That being said, we do expose the Signal class so you can patch your `.update()` onto the prototype if you wish.

**Reply by flmng0** [reply]
Thanks for your response.

I understand your rationale now. It's a very niche scenario, so I think I'll just add it in myself when it's needed.

Thanks again.
